src/main/java/org/reactome/qa/doisuggester/Main.java

```diff
@@ -301,64 +298,74 @@ private static void printOutput(Map<GKInstance, Set<Map<GKInstance, GKInstance>>
 
 	private static boolean hasNewInstanceEdit(List<GKInstance> currentRLEInstanceEdits, List<GKInstance> previousRLEInstanceEdits)
 	{
+		// start by assuming false and set to true when conditions require it.
+		boolean newInstanceEdit = false;
+		// I guess Justin assumed that if there are more InstanceEdits of the current version of an object, the it *must* be new.
+		// This is *probably* a safe assumption to make.
 		if (currentRLEInstanceEdits.size() != previousRLEInstanceEdits.size())
 		{
-			return true;
+			newInstanceEdit = true;
 		}
 		else
 		{
-			for (int i = 0; i < currentRLEInstanceEdits.size(); i++)
+			int i = 0;
+			while (i < currentRLEInstanceEdits.size() && !newInstanceEdit)
 			{
 				GKInstance currentIE = currentRLEInstanceEdits.get(i);
 				GKInstance previousIE = previousRLEInstanceEdits.get(i);
+				// It seems that in the case where current and previous versions have the same
+				// number of InstanceEdits, comparing DBIDs indicates if an InstanceEdit is new.
+				// I'm assuming Justin tested this theory more rigorously.
 				if (!currentIE.getDBID().equals(previousIE.getDBID()))
 				{
-					return true;
+					newInstanceEdit = true;
 				}
+				i++;
 			}
 		}
-		return false;
+		return newInstanceEdit;
 	}
 
 	private static GKInstance getGrandparentPathwayWithRLEInstanceEdit(GKInstance parentPathway, GKInstance newIE) throws Exception
 	{
+		GKInstance pathway = null;
 		List<GKInstance> grandparentPathways = (List<GKInstance>) parentPathway.getReferers(ReactomeJavaConstants.hasEvent);
-		if (grandparentPathways == null)
+		if (grandparentPathways != null)
 		{
-			return null;
-		}
-		// This never came up in my testing, but if it does, might need to be accounted
-		// for.
-		if (grandparentPathways.size() > 1)
-		{
-			System.out.println("TOO MANY PARENTS");
-			System.exit(0);
-		}
-		GKInstance grandparentPathway = grandparentPathways.get(0);
-		List<GKInstance> grandparentPathwayReviewedAuthoredRevisedInstances = grandparentPathway.getAttributeValuesList(ReactomeJavaConstants.reviewed);
-		grandparentPathwayReviewedAuthoredRevisedInstances.addAll(grandparentPathway.getAttributeValuesList(ReactomeJavaConstants.authored));
-		grandparentPathwayReviewedAuthoredRevisedInstances.addAll(grandparentPathway.getAttributeValuesList(ReactomeJavaConstants.revised));
-		boolean hasRLEInstanceEdit = false;
-		for (GKInstance grandparentPathwayIE : grandparentPathwayReviewedAuthoredRevisedInstances)
-		{
-			if (newIE.getDBID().equals(grandparentPathwayIE.getDBID()))
+			// This never came up in my testing, but if it does, might need to be accounted
+			// for.
+			if (grandparentPathways.size() > 1)
```

Comment:
This case, when it comes up, does need to be accounted for.  It would be proper to return all pathways which are at the highest level found to contain the RLE instance edit.

Reply:
That is not my comment, it's Justin's. He probably should have accounted for it already but I don't know how we would have planned to. I can try to account for that in future PRs.